### PR TITLE
New ad network: Mobusi. 

### PR DESCRIPTION
**Pull request by Mobusi-ads**
Hey guys,
we added our ad network "Mobusi" to the amp project. Everything was successfully tested.

I have signed CLA doc as josea.cano@mobusi.com, and in GitHub I am using same email account.

Best regards,
Jose A. cano.

### CODE REVIEW

**Comment by Mobusi-ads** [comment]
Cla doc by josea.cano@mobusi.com

captura

**Comment by Mobusi-ads** [comment]
Sorry closed accidentally

**Comment by Mobusi-ads** [comment]
I do not understand the problem with cla/google, as you can see I have signed the document with the email josea.cano@mobusi.com, which is the same email as the github account that performs the push request, am I missing some step that I don't know? Is there something I have not done correctly? Or is it a bug?

**Comment by jridgewell** [comment]
The commits were committed using the email josea.cano@impresionesweb.com. Did you code this on your local computer, or using GitHub's edit interface?

**Comment by jridgewell** [comment]
Try doing:

```
git config --global user.name "John Doe"
git config --global user.email johndoe@example.com
git rebase -i master -x "git commit --amend --reset-author -CHEAD"
```

**Comment by Mobusi-ads** [comment]
thx jridgewell

The pull requests are made with the github interface, the commit was done on the pc, (I could not remember that the global mail configured on that computer for git was josea.cano@impresionesweb.com).

Best regards.

**Comment by jridgewell** [comment]
Alright last part from me: You'll have to rebase off of current master to get the tests to pass. See https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#pull-the-latest-changes-from-the-amphtml-repository

File: ads/mobusi.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/mobusi.js]: 2017

File: ads/mobusi.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+import {writeScript} from '../3p/3p';
+
+export function mobusi(global, data) {
+  const adsrc = data.src;
+  const timestamp = (Date.now() % 2147483648) + Math.random();
+  if (typeof adsrc != 'undefined') {
+    writeScript(global, adsrc + '&lgid=' + timestamp);
```
Comment (zhouyx) [inline ads/mobusi.js]: use `encodeURIComponent(adsrc)`

File: ads/mobusi.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/mobusi.md]: 2017

File: ads/mobusi.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+# Mobusi
+
+## Example
+
+```html
+  <amp-ad width="300" height="450"
+    type="mobusi"
+    src="https://play.leadzu.com/banner_request.php?m=1GBTSITE40382X2&a=&idtm=300x450&lgid=324132413241234123">
+  <div placeholder></div>
```
Comment (zhouyx) [inline ads/mobusi.md]: We now provide default placeholder/fallback to every ad. Please only add them if you are asking for customized placeholder/fallback.

File: ads/mobusi.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript} from '../3p/3p';
+
+export function mobusi(global, data) {
+  const adsrc = data.src;
```
Comment (zhouyx) [inline ads/mobusi.js]: `data.src` should always exist? Please consider using `validateData()` function.

File: ads/mobusi.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  <div fallback>No ads</div>
+```
+
+## Configuration
+For semantics of configuration, please see ad network documentation.
```
Comment (zhouyx) [inline ads/mobusi.md]: Please link the ad network documentation here.

File: ads/mobusi.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+### Direct URL Call:
+* `src` - It is a custom URL for each area of the web of our publishers.
```
Comment (zhouyx) [inline ads/mobusi.md]: If `src` is required, please indicate.

File: examples/ads.amp.html [hunk 8]
```diff
@@ -840,6 +841,14 @@ <h2 id="mixpo">Mixpo</h2>
       data-subdomain = "www">
   </amp-ad>
 
+  <h2 id="mobusi">Mobusi</h2>
+
+  <amp-ad width="300" height="450"
+          type="mobusi"
+          src="https://play.leadzu.com/banner_request.php?m=1GBTSITE40382X2&a=&idtm=300x450&lgid=324132413241234123">
+    <div placeholder></div>
```
Comment (zhouyx) [inline examples/ads.amp.html]: no need for placeholder/fallback as default one is supported.

File: examples/ads.amp.html [hunk 9]
```diff
@@ -840,6 +841,14 @@ <h2 id="mixpo">Mixpo</h2>
       data-subdomain = "www">
   </amp-ad>
 
+  <h2 id="mobusi">Mobusi</h2>
+
+  <amp-ad width="300" height="450"
+          type="mobusi"
```
Comment (zhouyx) [inline examples/ads.amp.html]: please fix indentation.

File: ads/mobusi.js (hunk 1 above)
Comment (Mobusi-ads) [inline ads/mobusi.js]: Copy-paste bad habit.

:(

File: ads/mobusi.js (hunk 5 above)
Comment (Mobusi-ads) [inline ads/mobusi.js]: Currently yes, but later we will have several formats.

File: ads/mobusi.js (hunk 2 above)
Comment (Mobusi-ads) [inline ads/mobusi.js]: thx, changed.

File: examples/ads.amp.html (hunk 9 above)
Comment (Mobusi-ads) [inline examples/ads.amp.html]: Ok,changed

File: examples/ads.amp.html (hunk 8 above)
Comment (Mobusi-ads) [inline examples/ads.amp.html]: ok, changedd.

File: ads/mobusi.md (hunk 7 above)
Comment (Mobusi-ads) [inline ads/mobusi.md]: Currently yes, but it is something that we are going to change soon, allowing other amp-ad formats

**Comment by Mobusi-ads** [comment]
Forgive the delay in the changes, I was in a small vacation.

File: examples/ads.amp.html [hunk 10]
```diff
@@ -848,6 +849,12 @@ <h2 id="mixpo">Mixpo</h2>
       data-subdomain = "www">
   </amp-ad>
 
+  <h2 id="mobusi">Mobusi</h2>
+
+  <amp-ad width="300" height="450"
+    type="mobusi"
+    src="https://play.leadzu.com/banner_request.php?m=1GBTSITE40382X2&a=&idtm=300x450&lgid=324132413241234123">
+
```
Comment (zhouyx) [inline examples/ads.amp.html]: oops, missed `</amp-ad>`

File: ads/mobusi.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+-->
+
+# Mobusi
+
+## Example
+
+```html
+  <amp-ad width="300" height="450"
+    type="mobusi"
+    src="https://play.leadzu.com/banner_request.php?m=1GBTSITE40382X2&a=&idtm=300x450&lgid=324132413241234123">
```
Comment (zhouyx) [inline ads/mobusi.md]: missed `</amp-ad>`

File: ads/mobusi.js (hunk 5 above)
Comment (zhouyx) [inline ads/mobusi.js]: OK. I think it's always a good check for now. Optional though.

File: ads/mobusi.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  const timestamp = (Date.now() % 2147483648) + Math.random();
+  if (typeof adsrc != 'undefined') {
+    writeScript(global, encodeURIComponent(adsrc + '&lgid=' + timestamp));
```
Comment (zhouyx) [inline ads/mobusi.js]: you want the `&lgid=` and `timestamp to be encoded as well?

Comment (Mobusi-ads) [inline ads/mobusi.js]: As chained to join the same URL as adsrc, I think the right thing is that it also encode

File: ads/mobusi.md (hunk 11 above)
Comment (Mobusi-ads) [inline ads/mobusi.md]: upss!!

File: examples/ads.amp.html (hunk 10 above)
Comment (Mobusi-ads) [inline examples/ads.amp.html]: changed

File: ads/mobusi.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
+<!---
```
Comment (zhouyx) [inline ads/mobusi.md]: need to link this doc from amp-ad.md.

File: ads/mobusi.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+```html
+  <amp-ad width="300" height="450"
+    type="mobusi"
+    src="https://play.leadzu.com/banner_request.php?m=1GBTSITE40382X2&a=&idtm=300x450&lgid=324132413241234123">
+``</amp-ad>
+
+## Configuration
+For more information on configuration semantics, refer to the documentation 
+provided in the Mobusi panel for your users. Http://www.mobusi.com
```
Comment (zhouyx) [inline ads/mobusi.md]: Make the url a link please, maybe link the url from `documentation`?

File: examples/ads.amp.html [hunk 15]
```diff
@@ -848,6 +849,13 @@ <h2 id="mixpo">Mixpo</h2>
       data-subdomain = "www">
   </amp-ad>
 
+  <h2 id="mobusi">Mobusi</h2>
+
+  <amp-ad width="300" height="450"
+    type="mobusi"
+    src="https://play.leadzu.com/banner_request.php?m=1GBTSITE40382X2&a=&idtm=300x450&lgid=324132413241234123">
```
Comment (zhouyx) [inline examples/ads.amp.html]: src not available. 
Please check so that the example is a rendered ad.

File: ads/mobusi.js (hunk 12 above)
Comment (zhouyx) [inline ads/mobusi.js]: Sure, it really depends on how you decode the url on your side.

**Comment by zhouyx** [comment]
@Mobusi-ads Any update on the PR?

**Comment by zhouyx** [comment]
@Mobusi-ads Any update? I'll close the PR for now. Please feel free to reopen it later.

**Pull request closed without merging**